x/fswap/keeper/genesis.go

```diff
@@ -5,21 +5,47 @@ import (
 	"github.com/Finschia/finschia-sdk/x/fswap/types"
 )
 
-// InitGenesis initializes the module's state from a provided genesis
-// state.
-func (k Keeper) InitGenesis(ctx sdk.Context, genState types.GenesisState) {
-	if err := k.SetParams(ctx, genState.Params); err != nil {
-		panic(err)
+// InitGenesis initializes the module's state from a provided genesis state.
+func (k Keeper) InitGenesis(ctx sdk.Context, genState *types.GenesisState) error {
+	if err := genState.Validate(); err != nil {
+		return err
 	}
-	if err := k.SetSwapped(ctx, genState.Swapped); err != nil {
-		panic(err)
+
+	if len(genState.GetSwaps()) > k.config.MaxSwaps && !k.isUnlimited() {
+		return types.ErrCanNotHaveMoreSwap.Wrapf("cannot initialize genesis state, there are more than %d swapInits", k.config.MaxSwaps)
+	}
+
+	if len(genState.GetSwappeds()) > k.config.MaxSwaps && !k.isUnlimited() {
+		return types.ErrCanNotHaveMoreSwap.Wrapf("cannot initialize genesis state, there are more than %d swapped", k.config.MaxSwaps)
+	}
+
+	for _, swap := range genState.GetSwaps() {
+		if err := k.MakeSwap(ctx, swap); err != nil {
```

Comment:
`MakeSwap()` would initialize the corresponding statistics into the default value (zero). We must use dedicated logic (in this case, setxxx variants) in `InitGenesis()`. Also, for the swappeds.

Reply:
I fixed InitGenesis()